lib/game.rb

```diff
@@ -1,17 +1,84 @@
 # frozen_string_literal: true
 
-# class Game checks for the rows,columns and diagonals to check if anyone has won or else
-# tracks the count of turns and returns a draw if the bosrd is full before anyone wins.
+require_relative '../lib/board.rb'
+require_relative '../lib/ui.rb'
+
+# class Game Initiate a new board (using Board class) and status,
+# plays one match, and update the UI (using Ui module) and change
+# the player score in the mean time.
 class Game
-  def initialize
-    @count = 0
+  include Ui
+
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    new_match
+  end
+
+  private
+
+  def new_match
+    @player1.add_mark(nil)
+    @status = 'continue'
+    @rounds = 0
+    @board = Board.new
+    @marked_cells = []
+    @turn = @player2
+    assign_players_signs
+    play
+  end
+
+  def assign_players_signs
+    while @player1.mark.nil?
+      user_input = prompt("\nPlayer 1, Please choose a mark? (X/Y) ").upcase
+      break if %w[X O].include?(user_input)
+    end
+    @player1.add_mark(user_input)
+    @player2.add_mark(@player1.mark == 'X' ? 'O' : 'X')
+  end
+
+  def play
+    while @status == 'continue'
+      @turn = ([@player1, @player2] - [@turn]).first
+      prompt_cell
+      update
+    end
+    update_score
+    play_again?
+  end
+
+  def prompt_cell
+    cell = nil
+    while cell.nil?
+      answer = prompt("\nPlayer #{@marked_cells.length % 2 + 1} turn\nWhere would you like to put your mark?").to_i
+      cell = answer unless @marked_cells.include?(answer) || [1, 2, 3, 4, 5, 6, 7, 8, 9].include?(answer) == false
```

Comment:
With that approach, the Player won't know when it put the mark on a cell already filled.
Do you think to improve it, please? I'll be great !!!